### ISSUE TITLE
Double check the Averaging::evaluate function

### ISSUE THREAD

**Issue by bangerth** [geodynamics/aspect#1047]
Discovered in #1045: The `Averaging::evaluate()` function only does the averaging guarded by this check:
```
       /**
        * Check if the size of the viscosities (and thereby all the other vectors) is larger
        * than one. Averaging over one or zero points does not make a difference anyway,
        * and the normalized weighted distance averaging schemes need the distance between
        * the points and can not handle a distance of zero.
        */
       if (out.viscosities.size() > 1)
```
But checking the viscosities is a particularly poor choice since we document that one may suppress the computation of viscosities if the vector with input strain rates has size zero. I don't know if that means that the output vector of viscosities may also be zero, but it's definitely a poor choice to check. Any other variable may be a better one to test.

**Comment by lkaratun** [comment]
working on it

**Comment by lkaratun** [comment]
submitted the pull request [https://github.com/geodynamics/aspect/pull/1091](url)

**Comment by gassmoeller** [comment]
closed by #1091
